File: VehicleMonitoringSystemWeb/vms_frontend/src/models/employee.ts
export default class Employee {
  public id: string;
  public roleId: number;
  public companyId: number;
  public firstName: string;
  public lastName: string;
  public email: string;
  public password: string | undefined;

  constructor(id: string, roleId: number,
              companyId: number, firstName: string,
              lastName: string, email: string,
              password: string|undefined) {
    this.id = id;
    this.roleId = roleId;
    this.companyId = companyId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.password = password;
  }

  public getFullName () : string {
    return this.firstName + ' ' + this.lastName;
  }
}

File: VMS_Web/VMS_Web/Data/DatabaseModels/Employee.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VMS_Web.Data.DatabaseModels
{
    [Table("employee")]
    public class Employee
    {
        [Key]
        [Column("id")]
        public string Id { get; set; }

        [Column("role_id")] 
        public short RoleId { get; set; }
        [ForeignKey("RoleId")]
        public Role Role { get; set; }
        
        [Column("company_id")] 
        public int CompanyId { get; set; }
        [ForeignKey("CompanyId")]
        public Company Company { get; set; }       
        
        [Column("first_name")]
        [MaxLength(40)]
        public string FirstName { get; set; }
        
        [Column("last_name")]
        [MaxLength(40)]
        public string LastName { get; set; }
        
        [Column("email")]
        [MaxLength(40)]
        public string Email { get; set; }
    }
}